services/worker/src/worker/job_runners/split/duckdb_index.py

```diff
@@ -138,7 +141,22 @@ def compute_index_rows(
     logging.debug(CREATE_SEQUENCE_COMMAND)
     con.sql(CREATE_SEQUENCE_COMMAND)
 
-    create_command_sql = f"{CREATE_TABLE_COMMAND.format(columns=column_names)} read_parquet({parquet_urls});"
+    # see https://pypi.org/project/hf-transfer/ for more details about how to enable hf_transfer
+    os.environ["HF_HUB_ENABLE_HF_TRANSFER"] = "1"
+    for parquet_file in parquet_file_names:
+        hf_hub_download(
```

Comment:
For https://huggingface.co/datasets/bigcode/the-stack, this means downloading 2TB, no?

I think we have to check the size and stop downloading if the sum of the parquet files is more than 5GB.

Reply:
There is a validation already done here https://github.com/huggingface/datasets-server/pull/1717/files#diff-d9a2c828d7feca3b7f9e332e040ef861e842a16d18276b356461d2aa34396a8aR97 that will prevent processing splits with parquets bigger than 5 GB

Reply:
Oh OK! Thanks for clarifying